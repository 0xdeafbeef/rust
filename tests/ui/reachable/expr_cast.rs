//@ edition: 2024
#![deny(unreachable_code)]

fn a() {
    _ = {return} as u32; //~ error: unreachable
}

fn b() {
    (return) as u32; //~ error: unreachable
}

fn main() {}
