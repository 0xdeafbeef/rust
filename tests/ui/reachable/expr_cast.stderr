error: unreachable expression
  --> $DIR/expr_cast.rs:5:9
   |
LL |     _ = {return} as u32;
   |         ^------^^^^^^^^
   |         ||
   |         |any code following this expression is unreachable
   |         unreachable expression
   |
note: the lint level is defined here
  --> $DIR/expr_cast.rs:2:9
   |
LL | #![deny(unreachable_code)]
   |         ^^^^^^^^^^^^^^^^

error: unreachable expression
  --> $DIR/expr_cast.rs:9:5
   |
LL |     (return) as u32;
   |     --------^^^^^^^
   |     |
   |     unreachable expression
   |     any code following this expression is unreachable

error: aborting due to 2 previous errors

